app/update.go

```diff
@@ -17,22 +24,238 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		m.list.SetSize(msg.Width-h, msg.Height-v)
 
 	case tea.KeyMsg:
+		if m.list.SettingFilter() && key.Matches(msg, m.keys.yankFilter) {
+			//filterValue := m.list.FilterValue()
+
+			filterMatches := m.filterMatches()
+			if len(filterMatches) >= 1 {
+				err := clipboard.WriteAll(strings.Join(filterMatches, "\n"))
+				if err == nil {
+					return m, tea.Quit
+				}
+				cmds = append(
+					cmds,
+					m.list.NewStatusMessage(statusMessageStyle("Failed to copy all selected items.")),
+				)
+			}
+			return m, tea.Batch(cmds...)
+		}
+
 		// Don't match any of the keys below if we're actively filtering.
-		if m.list.FilterState() == list.Filtering {
+		if m.list.SettingFilter() {
+			break
+		}
+
+		i, ok := m.list.SelectedItem().(item)
+		if !ok {
+			switch {
+			case key.Matches(msg, m.keys.more):
+				m.list.SetShowHelp(!m.list.ShowHelp())
+				m.updatePaginator()
+			}
 			break
+
 		}
-		switch msg.String() {
-		case "p":
+		title := i.Title()
+		fullValue := i.TitleFull()
+		fp := i.FilePath()
+		desc := i.TimeStamp()
+
+		switch {
+
+		case key.Matches(msg, m.keys.choose):
+			selectedItems := m.selectedItems()
+			if len(selectedItems) < 1 && fp == "null" {
+				err := clipboard.WriteAll(fullValue)
+				utils.HandleError(err)
+				return m, tea.Quit
+			}
+
+			if len(selectedItems) >= 1 {
+				yank := ""
+				for _, item := range selectedItems {
+					if fullValue != item.Value {
+						yank += item.Value + "\n"
+					}
+				}
+				yank += fullValue
+				err := clipboard.WriteAll(yank)
+				if err == nil {
+					return m, tea.Quit
+				}
+				cmds = append(
+					cmds,
+					m.list.NewStatusMessage(statusMessageStyle("Failed to copy all selected items.")),
+				)
+			}
+
+			if fp != "null" {
+				ds := config.DisplayServer() // eg "wayland"
+				err := shell.CopyImage(fp, ds)
+				utils.HandleError(err)
+				return m, tea.Quit
+			}
+
+			if len(os.Args) > 2 {
+				if utils.IsInt(os.Args[2]) {
+					shell.KillProcess(os.Args[2])
+					return m, tea.Quit
+				}
+			} else if len(os.Args) > 1 {
```

Comment:
Assuming you applied my previous suggestion removing the else

```suggestion
			if len(os.Args) > 1 &&  os.Args[1] == "keep" {
```

Reply:
Above refactor should solve this also.